### PR TITLE
Ib sy db

### PR DESCRIPTION
**Pull request by Doobss**

### CODE REVIEW

File: client/appState/index.js [hunk 1]
```diff
@@ -6,23 +6,38 @@ export const DispatchContext = React.createContext([null, () => {}]);
...
+const initUser = {
+  userName: 'Tim',
+  lat: null,
+  lng: null,
+}
+
 const initAppState = {
-  dev: { logs: false },
+  dev: { logs: true },
   Auth: { AuthData: false, },
```
Comment (7socks) [inline client/appState/index.js:34]: Could you change this back to false by default?

File: client/app.js [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+const Footer = styled.footer`
+  height: 2em;
+  width: 100%;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  background-color: rgb(35, 35, 35);
+`
+
+export default App;
```
Comment (7socks) [inline client/app.js]: There seem to be two different app.js files, one in client and one in client/src. Not sure what's the intention here, but I'd like it to be in src.

File: client/api/index.js [hunk 3]
```diff
@@ -0,0 +1,115 @@
+
```
Comment (7socks) [inline client/api/index.js]: Reminder in case there are other files that depend on api.js -- update refs

File: client/app.js (hunk 2 above)
Comment (7socks) [reply client/app.js]: Also these styles have been moved, so you can probably delete this app.js file. It looks like its an older version.

File: server/DB/post.js [hunk 4]
```diff
@@ -18,14 +18,18 @@ POST.user = async (newUser) => {
...
   return await POST(newClaim, 'donated')
 }
 
+POST.session = async (newSession) => {
+  return await POST(newSession, 'sessions')
+}
+
 // const genRand = require('./generator')
 // const test = async () => {
 //   const res = await POST.user(genRand.user(23423))
```
Comment (7socks) [inline server/DB/post.js]: Marking commented out lines for removal later (or now, if possible)

File: server/DB/put.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+//       newData = [ newData ]
+//     }
+//     const res = await sql`UPDATE ${sql(table)} ${ sql(newData) }`
+//     console.log('PUT res for', table, res)
+//     return res
+//   } catch(err) {
+//     console.log('PUT err for', table, err.message, 'for ', newData)
+//     return {}
+//   }
+// }
```
Comment (7socks) [inline server/DB/put.js]: More commented lines to possibly remove

File: server/hydrate/generator.js [hunk 6]
```diff
@@ -39,31 +23,46 @@ const claimableLengths = {
...
+'she',
+'is',
+'not',
+'even',
+'looking',
+'im',
+'not',
+'a',
+'butt'
+]
```
Comment (7socks) [inline server/hydrate/generator.js:60]: Reminder that this project will be on our portfolios...

File: server/hydrate/generator.js [hunk 7]
```diff
@@ -233,6 +184,138 @@ module.exports.all = genAll
...
+//     posted_by: userId,
+//     title: titles[( randInt(titlesL) )],
+//     description: descriptions[( randInt(descriptionsL) )],
+//     category: categorys[( randInt(categorysL) )],
+//     tag: [ tags[( randInt(tagsL) )] ],
+//     charity_only: !!randInt(1, 'round'),
+//     lat,
+//     lng,
+//   }
+// }
```
Comment (7socks) [inline server/hydrate/generator.js:318]: WIP?

**Review by 7socks — CHANGES_REQUESTED** [review]

File: client/app.js (hunk 2 above)
Comment (Doobss) [reply client/app.js]: @7socks Yea Steph and I have rectified the double app.js thing, we also moved the appState and api folders into src.

**Review by positivefx — APPROVED** [review]

File: client/src/api/location.js [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+  //   return fetch(`https://api.geoapify.com/v1/ipinfo?apiKey=${process.env.IP_KEY}`)
+  //   .then(resp => resp.json())
+  //   .then((userLocationData) => {
+  //     locationLogs && console.log({userLocationData});
+  //     const { location } = userLocationData
+  //     return { lat: location.latitude, lng: location.longitude  }
+  //   })
+  //   .catch(err => locationLogs && console.log(`IP location fetch err ${err.message}`))
+  // }
+  // else {
```
Comment (7socks) [inline client/src/api/location.js:64]: marking commented lines

File: client/src/appState/reducers/user.js [hunk 9]
```diff
@@ -0,0 +1,26 @@
+
+
+
+
+function reducer(state, action) {
+
+  const { type, payload } = action
+  const { dev } = state
+  // console.log({ type, payload })
+  dev.logs && console.log('user REDUCER', state, type, payload)
```
Comment (7socks) [inline client/src/appState/reducers/user.js:10]: remove this console.log eventually

**Review by 7socks — APPROVED** [review]

**Pull request merged**